ADRs/0018 - SDValue.md

```diff
@@ -0,0 +1,92 @@
+# SDValue
+
+## Status
+**Discussion**
+
+Proposed by: Adam Gibson (8th Mar 2022)
+
+
+## Context
+
+Onnx and Tensorflow both support miscellaneous data structures to support the 
+building and execution of data flow graphs.
+
+Programming primitives such as lists, maps and optional types
+can be valuable in building data flow graphs allowing the developer to express a 
+wide variety of operations encoded within a neural network.
+
+Tensorflow uses [RaggedTensors](https://www.tensorflow.org/guide/ragged_tensor).
+Onnx uses [sequences](https://github.com/onnx/onnx/blob/main/docs/IR.md)
+Of note is onnx also supports optionals and maps.
+
+
+These primitives can actually be passed in as inputs and returned as outputs
+from a graph as well.
+
+In order to simplify the usage of these primitives a value type is used
+to indicate what type of value it is being passed in (maps, lists, tensors,..)
+
+
+SDValue is the Samediff interpretation of this and allows [TensorArrays](./nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/shape/tensorops/TensorArray.java) to be passed around. 
+as well as other types to be used within execution of a graph.
+
+
+
+## Proposal
+
+Create an SDValue class and associated enum abstraction for passing around and manipulating
+variables of different types.
+
+The following types will be supported:
+1. LIST
+2. DICT
+3. TENSOR
+
+
+An SDValue will be passed around similar to placeholders and are utilized within InferenceSession to 
+execute operations within a graph.
```

Comment:
Please use present tense in the Proposal section.

Also please ensure that the Markdown renders properly on Github. In particular, have a space after the final `#`, so headlines show up as actual headlines.